### PR TITLE
implement zrangebyscore and fix make error

### PR DESCRIPTION
**Pull request by piaoairy219**
1 implement zrangebyscore/zrevrangebyscore and its unit test, but just zrangebyscore was tested passed(will pull the tikvsdk new version in go.mod in some days laster and enable zrevrangebyscore)
2 zrange/zrevrange: fix a bug which may be cause repeating for ever , and some code style's small change and implement don't change in fact
3 fix the make errror if git log includes a '

### CODE REVIEW

File: db/zset.go [hunk 1]
```diff
@@ -259,6 +262,99 @@ func (zset *ZSet) ZAnyOrderRange(start int64, stop int64, withScore bool, positi
...
+	if offset < 0 || count == 0 {
+		return nil, nil
+	}
+
+	dkey := DataKey(zset.txn.db, zset.meta.ID)
+	scorePrefix := ZSetScorePrefix(dkey)
+
+	startPrefix := make([]byte, len(scorePrefix)+byteScoreLen)
+	copy(startPrefix, scorePrefix)
+	byteStartScore := EncodeFloat64(startScore)
```
Comment (nioshield) [inline db/zset.go]: assignment mismatch: 1 variable but EncodeFloat64 returns 2 values

File: db/zset.go [hunk 2]
```diff
@@ -259,6 +262,99 @@ func (zset *ZSet) ZAnyOrderRange(start int64, stop int64, withScore bool, positi
...
+	copy(startPrefix[len(scorePrefix):], byteStartScore)
+
+	stopPrefix := make([]byte, len(scorePrefix)+byteScoreLen)
+	copy(stopPrefix, scorePrefix)
+	byteStopScore := EncodeFloat64(stopScore)
```
Comment (nioshield) [inline db/zset.go]: assignment mismatch: 1 variable but EncodeFloat64 returns 2 values

**Comment by shafreeck** [comment]
@piaoairy219 Please fix the compiling errors

**Comment by shafreeck** [comment]
There is no response for a long time, I prefer to merge this PR first and then fix the issues introduced.

**Pull request merged**